src/mini_trainer/train.py

```
@@ -719,17 +981,45 @@
-            "WORLD_SIZE": world_size
+            "WORLD_SIZE": world_size,
         }
+        
+        # Initialize wandb with the same params config
+        use_wandb = wandb_project is not None
+        if use_wandb:
+            # we rely on the WANDB_API_KEY being set as our primary mechanism for
+            # authentication. So we error out here if it was requested but the user
+            # is not authenticated
+            if os.environ.get("WANDB_API_KEY") is None:
+                raise ValueError("WANDB_API_KEY is not set. Please set the WANDB_API_KEY environment variable.")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)